lib/build.js

```diff
@@ -214,38 +214,43 @@ class BuildModel extends BaseModel {
      * @return {Promise}
      */
     update() {
-        const dirty = this.isDirty('status');
-
-        return super.update()
-            .then(() => {
-                if (!dirty) {
-                    return this;
-                }
+        let prom = Promise.resolve();
+
+        // check if the status is changing
+        if (this.isDirty('status')) {
+            // stop the build if we're done
+            if (this.isDone()) {
+                prom = prom
+                    .then(() => this.stop());
+            }
+            // update scm with status
+            prom = prom
+                .then(() => this.pipeline)
+                .then(pipeline => this.updateCommitStatus(pipeline));
+        }
 
-                // update scm with status
-                return this.pipeline
-                    .then(pipeline => this.updateCommitStatus(pipeline))
-                    .then(() => this);
-            });
+        return prom
+            .then(() => super.update())
+            .then(() => this);
     }
 
     /**
-     * Stop a build and update github status as failure
+     * Stop a build
```

Comment:
So that stop just stops the build, update is the funnel for updating github status.